specificatie/zoek-personen.yaml

```diff
@@ -0,0 +1,313 @@
+        '403':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/403"
+        '406':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/406"
+        '500':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/500"
+        '501':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/501"
+        '503':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/503"
+        'default':
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.3.0/api-specificatie/common.yaml#/components/responses/default"
+      tags:
+        - Ingeschreven Personen
+components:
+  schemas:
+    ZoekPersonenQuery:
+      type: object
+      required:
+        - type
+        - fields
+      discriminator:
+        propertyName: type
+        mapping:
+          ZoekMetGeslachtsnaamEnGeboortedatum: '#/components/schemas/ZoekMetGeslachtsnaamEnGeboortedatum'
+          ZoekMetGeslachtsnaamEnGemeenteVanInschrijving: '#/components/schemas/ZoekMetGeslachtsnaamEnGemeenteVanInschrijving'
+          ZoekMetBurgerservicenummer: '#/components/schemas/ZoekMetBurgerservicenummer'
+          ZoekMetPostcodeEnHuisnummer: '#/components/schemas/ZoekMetPostcodeEnHuisnummer'
+          ZoekMetStraatHuisnummerEnGemeenteVanInschrijving: '#/components/schemas/ZoekMetStraatHuisnummerEnGemeenteVanInschrijving'
+          ZoekMetNummeraanduidingIdentificatie: '#/components/schemas/ZoekMetNummeraanduidingIdentificatie'
+      properties:
+        type:
+          type: string
+        fields:
+          $ref: 'filter.yaml#/components/schemas/Fields'
+        inclusiefOverledenPersonen:
+          description: |
+            Als je ook overleden personen in het antwoord wilt, geef dan de parameter inclusiefOverledenPersonen op met waarde True.  Zie [functionele specificaties](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.3.0/features/overleden_personen.feature)
+          type: boolean
+          example: true
+    ZoekMetBurgerservicenummer:
+      required:
+        - burgerservicenummer
+      allOf:
+        - $ref: '#/components/schemas/ZoekPersonenQuery'
+        - type: object
+          properties:
+            burgerservicenummer:
+              type: array
+              items:
+                $ref: 'ingeschrevenpersoon.yaml#/components/schemas/Burgerservicenummer'
+              minItems: 1
+              maxItems: 10
+    ZoekMetGeslachtsnaamEnGeboortedatum:
```

Comment:
in elke variant zijn de mogelijke parameters gelijk, alleen required verschilt. Zou je dan ook alle parameters in ZoekPersonenQuery kunnen definiëren, en in de zoekvariant alleen required zetten?

Reply:
Dat kan, maar daarmee vergroot je de aantal combinaties waar je geen resultaat krijgt. Waarom zou je bijv. zoeken mogelijk maken op bsn en nummeraanduidingidentificatie of zoeken op postcode, huisnummer en straatnaam. Verder vergroot je hiermee de attack service. Hoe meer combinaties je mogelijk maak door de optionele parameters, hoe groter de kans dat er een combinatie is waarmee je de beschikbaarheid van de API omlaag kan brengen.

Reply:
Ik ben al bezig met een aanzet voor de feature files. Zie zoek-met-bsn.feature en zoek-met-geslachtsnaam-geboortedatum.feature in de branch. Het verschilt qua opzet tov de bestaande feature files, omdat ik de feature files wil automatiseren zodat het kan worden gebruikt om de functionaliteit te valideren